File: Programmers_cs/나누어 떨어지는 숫자 배열.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ConsoleApplication1
{
    class Program
    {
        static void Main(string[] args)
        {
            Solution so = new Solution();
            int[] arr = new int[] { 3, 2, 6 };
            foreach (int i in so.solution(arr, 10))
            {
                Console.WriteLine(i);
            }
        }
    }

    public class Solution
    {
        public int[] solution(int[] arr, int divisor)
        {
            List<int> vallist = new List<int>();
            int[] answer = new int[] { };
            foreach (int val in arr)
            {
                if (val % divisor == 0) vallist.Add(val);
            }
            if (vallist.Count == 0) vallist.Add(-1);
            answer = vallist.ToArray();
            Array.Sort(answer);
            return answer;
        }
    }
}

File: Programmers_cpp/나누어 떨어지는 숫자 배열.cpp
#include <iostream>
#include <string>
#include <vector>
#include <algorithm>

using namespace std;

vector<int> solution(vector<int> arr, int divisor) {
	vector<int> answer;

	for (int i = 0; i < arr.size(); i++) {
		if (arr[i] % divisor == 0) answer.push_back(arr[i]);
	}
	if (answer.size() == 0) answer.push_back(-1);

	sort(answer.begin(), answer.end());

	return answer;
}

int main(void) {
	vector<int> answer = solution({5, 9, 7, 10}, 5);

	//cout << answer;

	for (int i = 0; i < answer.size(); i++) {
		cout << answer[i];
	}
}
